tools/ci/build-test-macos-with-sanitizers.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This script builds iceoryx_hoofs und iceoryx_posh and executes all tests
+
+set -e
+
+DIR="$(cd -P "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+msg() {
+    printf "\033[1;32m%s: %s\033[0m\n" ${FUNCNAME[1]} "$1"
+}
+
+# go to repo root
+ROOT=${DIR}/../..
+cd ${ROOT}
```

Comment:
In `iceoryx_build_test.sh` dkroenke introduced this `WORKSPACE=$(git rev-parse --show-toplevel)`. This is more robust since the scripts can be moved around and will still work

Reply:
done